### DISCUSSION TITLE
Vision Support

### DISCUSSION THREAD

**Discussion by The-Michael-Chen** [CJCrafter/ChatGPT-Java-API#43] [General]
Hey! Is there support for GPT4 vision?

**Comment by CJCrafter** [comment]
I haven't specifically tried using a GPT4V model, but it uses the Assistants API, which we have support for. Assuming GPT4V simply uses any image file uploaded with a message in a thread, you can add it as a file to your message: https://github.com/CJCrafter/ChatGPT-Java-API/blob/master/src/main/kotlin/com/cjcrafter/openai/threads/message/CreateThreadMessageRequest.kt#L74

P.S. Hello from across the Charles! :+1:

**Reply by CJCrafter** [reply]
If you do any experimentation, consider making a PR to contribute to the examples
